packages/next-auth/src/utils/nextjs-base-path.ts

```diff
@@ -0,0 +1,15 @@
+import type { NextURL } from "next/dist/server/web/next-url";
+
+export default function (nextUrl: NextURL): string {
```

Comment:
Why do you use this function? NextUrl now contain  basePath prop. It's more transparent and supported way to extract basePath in next.js based projects. Are you agree? Or i miss something?

Reply:
Also, i think pathname is not an optimal option, let me explain via example:

1. Go to protected page like /orders/[orderId]/view and concrete /orders/1234/view
2. Redirect to signIn page
3. see callbackUrl like callbackUrl=/orders/[orderId]/view!!! Instead of /orders/1234/view

Reply:
> Why do you use this function? NextUrl now contain basePath prop. It's more transparent and supported way to extract basePath in next.js based projects. Are you agree? Or i miss something and didn't cover all cases?

Pretty good catch. I adapted to req.nextUrl.basePath

Reply:
@psperber can't reproduce step3 or don't understand what i mean?

I mean: after redirect from protected url to signIn page, value contained previous url of protected page will write into query param as invalid value, without replaced value. But after few test a can't reproduce this in my implementation #5491 . Sorry it's my fault

Reply:
> How about using `parseUrl(process.env.NEXTAUTH_URL).path` ? It should work fine, I remember tested it when I first review this PR

Use it for what? The problem with `parseUrl(process.env.NEXTAUTH_URL).path` is, that it contains the nextjs basepath. The nextUrl.pathname does not. The logic about publicPaths in line 115 and 116 are based on the nextUrl.pathname, which does not contain the nextjs basepath.
I think the current state it be least complex variant to do the job